lib/rubygems/source.rb

```diff
@@ -67,7 +67,9 @@ def <=>(other)
 
       return -1 if !other.uri
 
-      @uri.to_s <=> other.uri.to_s
+      return 1 unless @uri.to_s == other.uri.to_s
+
+      0
```

Comment:
Can you move this ↑ into a comment above line 70?

It'll make it harder to accidentally revert.